solana/pyth2wormhole/client/src/main.rs

```diff
@@ -301,56 +272,181 @@ async fn handle_attest(
+            "Attestation config (includes mapping accounts):\n{:#?}",
+            attestation_cfg
+        );
 
-    info!("Spinning up attestation sched jobs");
+        info!(
+            "{} symbol groups read, dividing into batches",
+            attestation_cfg.symbol_groups.len(),
+        );
 
-    let results = futures::future::join_all(attestation_sched_futs).await; // May never finish for daemon mode
+        let mut batches: Vec<_> = attestation_cfg
+            .symbol_groups
+            .clone()
+            .iter()
+            .map(|g| {
+                let conditions4closure = g.conditions.clone();
+                let name4closure = g.group_name.clone();
+
+                info!("Group {:?}, {} symbols", g.group_name, g.symbols.len(),);
+
+                // Divide group into batches
+                g.symbols
+                    .as_slice()
+                    .chunks(config.max_batch_size as usize)
+                    .map(move |symbols| {
+                        BatchState::new(name4closure.clone(), symbols, conditions4closure.clone())
+                    })
+            })
+            .flatten()
+            .enumerate()
+            .map(|(idx, batch_state)| (idx + 1, batch_state))
+            .collect();
+        let batch_count = batches.len();
+
+        // For enforcing min_msg_reuse_interval_ms, we keep a piece of
+        // state that creates or reuses accounts if enough time had
+        // passed
+        let message_q_mtx = Arc::new(Mutex::new(P2WMessageQueue::new(
+            Duration::from_millis(attestation_cfg.min_msg_reuse_interval_ms),
+            attestation_cfg.max_msg_accounts as usize,
+        )));
+
+        // Create attestation scheduling routines; see attestation_sched_job() for details
+        let mut attestation_sched_futs = batches.into_iter().map(|(batch_no, batch)| {
+            attestation_sched_job(AttestationSchedJobArgs {
+                batch,
+                batch_no,
+                batch_count,
+                n_retries,
+                retry_interval,
+                daemon,
+                rpc_cfg: rpc_cfg.clone(),
+                p2w_addr,
+                config: config.clone(),
+                payer: Keypair::from_bytes(&payer.to_bytes()).unwrap(),
+                message_q_mtx: message_q_mtx.clone(),
+            })
+        });
 
-    info!("Got {} results", results.len());
+        let joined_sched_futs = futures::future::join_all(attestation_sched_futs);
 
-    // With daemon mode off, the sched jobs return from the
-    // join_all. We filter out errors and report them
-    let errors: Vec<_> = results
-        .iter()
-        .enumerate()
-        .filter_map(|(idx, r)| {
-            r.as_ref()
-                .err()
-                .map(|e| format!("Error {}: {:#?}\n", idx + 1, e))
-        })
-        .collect();
+        info!("Spinning up attestation sched jobs");
+        if daemon {
+            let mut hasher = DefaultHasher::new();
```

Comment:
I think you want a cryptographic hash here -- if you get a hash collision, the code below isn't going to do the right thing.